### ISSUE TITLE
GET_FILE_ARRAY/GET_DIRECTORY_ARRAY: not working with files of size > 1 GB

### ISSUE THREAD

**Issue by Argent77** [WeiDUorg/weidu#119]
Both `GET_FILE_ARRAY` and `GET_DIRECTORY_ARRAY` are causing mod installations to fail when they encounter files of size >= 1 GB in their search path.

Example:
```
// Code:
GET_FILE_ARRAY files ~dlc~ ~.+\.zip~

// Resulting error message:
ERROR: Unix.Unix_error(67, "stat", "dlc/sod-dlc.zip")
```
Size of `sod-dlc.zip` is about 1.8 GB.

**Comment by FredrikLindgren** [comment]
Yes, there appears to be a number of remaining uses of unix_stat that could potentially fail on large files. I'll try to look it over during the week.

**Comment by Argent77** [comment]
Just a note: 64-bit binaries appear to work correctly with files > 1GB using unix_stat. However, 32-bit binaries do not (except on macOS).
